lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -397,48 +431,101 @@ LogicalResult AffineToPipeline::createPipelinePipeline(
 
   // Maintain mappings of values in the loop body and results of stages,
   // initially populated with the iter args.
-  BlockAndValueMapping valueMap;
-  for (size_t i = 0; i < iterArgs.size(); ++i)
-    valueMap.map(forOp.getBody()->getArgument(i),
-                 pipeline.getStagesBlock().getArgument(i));
+  BlockAndValueMapping iterValueMap;
+  for (size_t i = 0; i < forOp.getBody()->getNumArguments(); ++i)
```

Comment:
Does changing from `iterArgs.size()` to `forOp.getBody()->getNumArguments()` have any effect?